qiskit_nature/problems/second_quantization/lattice/__init__.py

```diff
@@ -58,3 +58,14 @@
 )
 from .models import FermiHubbardModel, IsingModel
 from .lattice_model_problem import LatticeModelProblem
+from ....deprecation import warn_deprecated, DeprecatedType, NatureDeprecationWarning
+
+warn_deprecated(
+    "0.5.0",
+    old_type=DeprecatedType.PACKAGE,
+    old_name="qiskit_nature.problems.second_quantization.lattice",
```

Comment:
I think we need to differentiate here:
- the `LatticeModelProblem` class did indeed go to `second_q.problems`
- but the `lattice` submodule will go to `second_q.hamiltonians`